tests/unit/test_endpoint_provider.py

```diff
@@ -433,3 +434,24 @@ def test_auth_schemes_conversion_first_authtype_unknown(
     at, sc = empty_resolver.auth_schemes_to_signing_ctx(auth_schemes)
     assert at == 'bar'
     assert sc == {'region': 'ap-south-2', 'signing_name': 's3'}
+
+
+@pytest.fixture(scope="class")
+def test_endpoint_provider(endpoint_provider, request):
+    request.cls.endpoint_provider = endpoint_provider
+
+
+@pytest.mark.usefixtures("test_endpoint_provider")
+class EndpointProviderCacheTestCase(unittest.TestCase):
+    @unittest.mock.patch("botocore.endpoint_provider.RuleSet.evaluate")
+    def test_endpoint_resolution_caches(self, mock_evaluate):
+
+        for _ in range(5):
+            self.endpoint_provider.resolve_endpoint(**{"Region": "us-east-1"})
```

Comment:
Why not the simpler:

```suggestion
            self.endpoint_provider.resolve_endpoint(Region="us-east-1")
```

(And the same a few lines below?)

Reply:
Totally. Not sure why I did it like this. I tried to make the change to `assert_called_once_with` but it seems as though it expects it in the dict form